processor/src/main/java/org/mapstruct/ap/internal/model/source/TargetReference.java

```diff
@@ -142,25 +148,26 @@ public TargetReference build() {
             // 3. @MappingTarget, with
             // 4. or without parameter name.
             String[] targetPropertyNames = segments;
-            List<PropertyEntry> entries = getTargetEntries( resultType, targetPropertyNames, false );
+            List<PropertyEntry> entries = getTargetEntries( resultType, targetPropertyNames );
```

Comment:
`getTargetEntries` .. I still think that `resolveTargetEntries` is better.. There is some analysis involved and its not a plain getter..